quickstart-example/frontend/src/App.tsx

```diff
@@ -1,66 +1,111 @@
-import { useEffect, useState } from "react";
-import { Wallet } from "fuels";
+import React, { useEffect, useState } from "react";
+import { bn } from "fuels";
+import "@fuel-wallet/sdk";
 import "./App.css";
 // Import the contract factory -- you can find the name in index.ts.
 // You can also do command + space and the compiler will suggest the correct name.
 import { CounterContractAbi__factory } from "./contracts";
 
 // The address of the contract deployed the Fuel testnet
 const CONTRACT_ID =
-  "0x2c38161e42ce14abdaa7950c04b13aa340eb67cf196c94b971d5f0175417f4f4";
+  "0x3edb96c23766b8504caaff042994efa18460e7ba27f60191394a6bcf5be8d7d8";
 
-//the private key from createWallet.js
-const WALLET_SECRET =
-  "0x07960a5124cd6e52b595aed727d640aed2e32cc587a34c09f1b48615378df752";
-
-// Create a Wallet from given secretKey in this case
-// The one we configured at the chainConfig.json
-const wallet = Wallet.fromPrivateKey(
-  WALLET_SECRET,
-  "https://beta-3.fuel.network/graphql"
-);
+function App() {
+  const [connected, setConnected] = useState<boolean>(false);
+  const [account, setAccount] = useState<string>("");
+  const [counter, setCounter] = useState<number>(0);
+  const [loaded, setLoaded] = useState(false);
+  
+  useEffect(() => {
+    setTimeout(() => {
+      checkConnection();
+      setLoaded(true);
+    }, 200)
+    if (connected) getCount();
+  }, [connected])
 
-// Connects out Contract instance to the deployed contract
-// address using the given wallet.
-const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
+  async function connect() {
+    if (window.fuel) {
+     try {
+       await window.fuel.connect();
+       const accounts = await window.fuel.accounts();
+       setAccount(accounts[0]);
+       setConnected(true);
+     } catch(err) {
+       console.log("error connecting: ", err);
+     }
+    }
+   }
 
-function App() {
-  const [counter, setCounter] = useState(0);
-  const [loading, setLoading] = useState(false);
+  async function checkConnection() {
+    if (window.fuel) {
+      const isConnected = await window.fuel.isConnected();
+      if (isConnected) {
+        const accounts = await window.fuel.accounts();
+        setAccount(accounts[0]);
+        setConnected(true);
+      }
+    }
+  }
 
-  useEffect(() => {
-    async function main() {
-      // Executes the counter function to query the current contract state
-      // the `.get()` is read-only, because of this it don't expand coins.
+  async function getCount() {
+    if (window.fuel) {
+      const wallet = await window.fuel.getWallet(account);
+      const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
       const { value } = await contract.functions.count().get();
-      setCounter(Number(value));
+      setCounter(Number(bn(value)));
```

Comment:
Do we need this bn around the value? If the value returned is a BN you can use toNumber().

Reply:
good suggestion, thanks!